src/python/espressomd/CMakeLists.txt

```diff
@@ -102,7 +99,9 @@ foreach(cython_file ${cython_SRC})
           set_source_files_properties(${outputpath} PROPERTIES COMPILE_FLAGS "-Wno-pedantic -Wno-unused-variable")
       endif()
       set_target_properties(${basename} PROPERTIES CXX_CLANG_TIDY "")
-      target_link_libraries(${basename} PRIVATE EspressoConfig EspressoCore EspressoScriptInterface)
+      target_link_libraries(${basename} PRIVATE EspressoConfig EspressoCore EspressoScriptInterface ${PYTHON_LIBRARIES})
+      target_include_directories(${basename} PRIVATE SYSTEM ${PYTHON_INCLUDE_DIRS} ${NUMPY_INCLUDE_DIR})
```

Comment:
Or add the `-undefined dynamic_lookup` flag here exclusively to the Python modules. It seems like that is what most Python modules do. For some reason, the official Python recommendation is to not link modules against libpython.

Reply:
Yes, that has the advantage that you always get the lib for the python executable which you are actually running. This needs binary compatibility, so it is probably a good idea.